### PR TITLE
25 create first dialog page

### LINKED ISSUE #25: Create first dialog page
Create the first dialog page based on the given design.

Figma:
https://www.figma.com/file/ZNuZoa1I6pRWBgRLAvRWNR/Prototyp?type=design&node-id=0-1&mode=design&t=rsxRwOtZIDo4q59s-0

### PR DESCRIPTION
**Pull request by mastoeppler**

### CODE REVIEW

File: public/data/dialog.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          "text": "Natürlich, Zog! Aber wie kann ich dir helfen? Ich habe noch nie ein Raumschiff gesehen!"
+        },
+        {
+          "speaker": "right",
+          "text": "Keine Sorge, Emma. Ich habe eine Karte, die uns zeigt, was wir brauchen. Lass uns gemeinsam auf ein Abenteuer gehen!",
+          "image": "/images/rectangle.png"
+        }
+      ]
+    },
+    {
```
Comment (DorinaSobiecki) [inline public/data/dialog.json]: It should be one .json per conversation

File: app/components/speechBubble/speechBubble.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const SpeechBubble: React.FC<SpeechBubbleProps> = ({ text, direction }) => {
+  const bubbleStyles = {
+    alignSelf: direction === 'left' ? 'flex-start' : 'flex-end',
+    backgroundColor: direction === 'left' ? '#fff' : '#F3F3F3',
+    color: direction === 'left' ? '#000' : '#333',
+  };
+
+  return (
+    <div className={`speech-bubble ${direction}`} style={bubbleStyles}>
```
Comment (DorinaSobiecki) [inline app/components/speechBubble/speechBubble.tsx:17]: As you already defined a CSS class for the direction the upper bubbleStyles can be in, these CSS classes

File: pages/dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    } else {
+      const nextSceneIndex = currentSceneIndex + 1;
+
+      if (nextSceneIndex < dialogData.story.length) {
+        setCurrentSceneIndex(nextSceneIndex);
+        setCurrentDialogIndex(0);
+        setDialog(dialogData.story[nextSceneIndex].dialog);
+        setImages(dialogData.story[nextSceneIndex].images[0]);
+      } else {
+        console.log('Story is finished');
```
Comment (DorinaSobiecki) [inline pages/dialog.tsx]: Remove log

File: pages/dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    }
+  };
+
+  const handleAnswerSelect = (isCorrect: boolean) => {
+    if (isCorrect) {
+      setTimeout(() => {
+        handleNext();
+      }, 1000); // 1 second delay
+    } else {
+      console.log('Wrong answer');
```
Comment (DorinaSobiecki) [inline pages/dialog.tsx]: remove log

File: pages/dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        leftCharacter={
+          <CharacterImage
+            hairColor={selectedHairColorCode}
+            hairType={selectedHair}
+            skinColor={selectedSkinColorCode}
+          />
+        }
+        rightCharacter={images.rightCharacter}
+        dialog={dialog[currentDialogIndex]}
+        actionButton={<ActionButton onClick={handleNext} />}
```
Comment (DorinaSobiecki) [inline pages/dialog.tsx]: Should be placed inside the layout

File: pages/dialog.tsx
Comment (DorinaSobiecki) [inline pages/dialog.tsx:1]: Outsource the logic into the layout so we don't have to repeat the logic for every conversation when we use separated JSONs

File: public/data/dialog.json
Comment (DorinaSobiecki) [inline public/data/dialog.json:1]: Please rename folder to "dialog"

**Review by DorinaSobiecki — CHANGES_REQUESTED** [review]
Take a look at my comments

File: pages/dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      const hair = localStorage.getItem('selectedHair') || '';
+      const skinColorCode = localStorage.getItem('selectedSkinColorCode') || '';
+
+      setSelectedHairColorCode(hairColorCode);
+      setSelectedHair(hair);
+      setSelectedSkinColorCode(skinColorCode);
+
+      loadStoredValues();
+    }
+  }, []);
```
Comment (SebastianMaier03) [inline pages/dialog.tsx]: Adapt Logic and fix dilaog so the character get loadet on empty storage

**Review by SebastianMaier03 — COMMENTED** [review]
Are there any new Uncommited changes or Updates?

**Review by SebastianMaier03 — CHANGES_REQUESTED** [review]
Remove Code that breaks default character loading.
Look at Figma and implement the hint and fix css.
As seen in Figma Wrong answer shoul change color to a different red after the hint was displayed.

**Comment by mastoeppler** [comment]
Still to do Hint and Question Design

**Comment by mastoeppler** [comment]
please check if the questions and characters are displayed correctly

**Review by DorinaSobiecki — CHANGES_REQUESTED** [review]
The speech bubbles are still quite high when there is less text in them.

The image is rendered very small

After a wrong answer to a question, a hit is displayed, after selecting the correct answer, the hint disappears and the question is displayed again for about 1sec before the dialog continues, please fix it so that after a hint is displayed and the correct answer is selected, the dialog continues without the question displayed again.

**Comment by mastoeppler** [comment]
I have revised it, please check it again

**Review by SebastianMaier03 — CHANGES_REQUESTED** [review]
grafik

Speach bubble still so far on top, Img size should be dynamic , and take a look at figma

**Pull request merged**